### DISCUSSION TITLE
GCC version for C++11

### DISCUSSION THREAD

**Discussion by amaroyo** [robotology/community#229] [Q&A] [answered]
Hi, 

I am using macOS and I was using the function ```std::to_string(myDoubleVar);``` which is working fine. However, I was compiling it in the linux setup and it's not recognized **'to_string' is not a member of 'std'** 

It seems that the function is C++11, but I thought we were using already the [C++11](https://github.com/robotology/QA/issues/173). 

Any ideas? Do we need to upgrade the linux setup with a newer version? 

Thanks, 
Alex.

**Comment by pattacini** [comment, marked as answer]
Hi @amaroyo,

C++11 is not enabled by default on Linux yet, or at least on certain flavors/versions of Linux; therefore, you have to explicitly tell CMake to compile the code with C++11 features enabled.

There are [multiple ways](https://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake) of doing this. If you have CMake 3.1, try the following directive:

```cmake
set(CMAKE_CXX_STANDARD 11)
```

**Comment by traversaro** [comment]
C++11 (actually C++14) will be the default from GCC 6.1 (https://gcc.gnu.org/projects/cxx-status.html#cxx14). The first Ubuntu version with it I think is Ubuntu 16.10 .
